apps/web/src/data/contract/chains.ts

```diff
@@ -9,7 +9,8 @@ import { CHAIN_ID } from 'src/typings'
 
 const MAINNET_CHAINS = [mainnet]
 
-const TESTNET_CHAINS = [goerli, optimismGoerli, baseGoerli]
+// Mainnet is required here due to hooks like useEnsData that only pull data from mainnet
+const TESTNET_CHAINS = [mainnet, goerli, optimismGoerli, baseGoerli]
```

Comment:
hmm, is there a better way to specify the fallbacks on ENS lookups to mainnet that doesn't involved adding it as a specified testnet chain? I feel like I expect this array to be the list of chains that we want the app enabled to interact with on a testnet environment. what are your thoughts?

Reply:
imo the testnet app is interacting with mainnet to pull ens data so this is valid. the `PUBLIC_DEFAULT_CHAINS` is the list of user facing chains that we display in the ui. how I see it there are three ways we can approach this:

- leave as is and only our own chain switching component to ensure mainnet is never displayed as an option to the user. (how the app is currently set up)
- pull goerli ens data when on the testnet deployment
- rework the ens hook to pull the data using a provider like `getEnsAddress` and `getEnsName` unsure if this works for avatar data though but I can double check.

let me know what you think would work best @nadiakiamilev

Reply:
nice! ok yeah I see the issue now. Sure happy to leave it as is then, considering that we want to be able to convey ens names regardless of environment! 

I like the third point! It feels like it separates the concern of ens data regardless of environment (which conceptually also seems to be the case that we want since we always want to rely on mainnet ens names). 

Don't want to block though and happy to go current direction! We can always re-visit.